skip validating goreleaser state when running release-dry-run on branch

on branches `make release-dry-run` should be with `GORELEASER_SKIP_VALIDATE=true`

```
   • releasing...             
   • loading config file       file=.goreleaser.yaml
   • loading environment variables
   • getting and validating git state
      • releasing v0.9.1, commit 1ef2bb8d28f5a7e38451c2ae605ba8bbc763667e
   ⨯ release failed after 0.15s error=git tag v0.9.1 was not made against commit 1ef2bb8d28f5a7e38451c2ae605ba8bbc763667e
make: *** [release-dry-run] Error 1
make/releasing.mk:39: recipe for target 'release-dry-run' failed
Error: Process completed with exit code 2.
```

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -26,7 +26,11 @@ jobs:
           echo ::set-output name=MAINNET::$(./script/mainnet-from-tag.sh ${GITHUB_REF#refs/tags/})
       - run: echo "Building for MAINNET=${{ steps.branch_name.outputs.MAINNET }}"
       - uses: actions/setup-go@v2
-      - name: release dry-run
+      - if: (github.ref == 'refs/heads/master') || (github.ref == 'refs/heads/mainnet/main')
+        name: release dry-run GORELEASER_SKIP_VALIDATE=true
+        run: make release-dry-run GORELEASER_SKIP_VALIDATE=true
+      - if: startsWith(github.ref,'refs/tags/v')
+        name: release dry-run
         run: make release-dry-run
       - if: startsWith(github.ref,'refs/tags/v') && github.repository == 'ovrclk/akash'
         name: setup release environment

